### ISSUE TITLE
XMLProducer fails when model contains a BEASTInterface object without an ID

### ISSUE THREAD

**Issue by tgvaughan** [CompEvol/beast2#421]
This means that logs, or anything that includes a fragment of the model XML in its output, can sometimes contain no output (or simply the word "null").

Maybe this isn't often a problem in practice as BEAUti automatically gives objects IDs when they aren't explicitly provided, but it is very easy for this to happen in hand-crafted XML that otherwise seems perfectly fine.

Maybe XMLParser itself should add dummy IDs to objects that don't have them?

**Comment by rbouckaert** [comment]
The test just added seems to succeed without problem: it creates a Nucleotide object without ID, then runs XMLProducer.toXML() on it. Perhaps there are some more conditions required for this to fail?

**Comment by tgvaughan** [comment]
Hrm, this is what happens when I don't specify a test with an issue.  I just poked around to see whether I'd accidentally included a fix -- I did have a fix at some point but wasn't game to commit as I was worried it could break something, since some code actually uses the null IDs to alter behaviour (eg log file headers) -- so it's possible.  Can't find anything though!  Maybe close for now and we'll reopen if the problem comes back.
